FirebaseDatabase/Sources/Core/FRepo.m

```diff
@@ -525,11 +525,13 @@ - (void)getData:(FIRDatabaseQuery *)query
     FQuerySpec *querySpec = [query querySpec];
     id<FNode> node = [self.serverSyncTree getServerValue:[query querySpec]];
     if (node != nil) {
-        block(nil, [[FIRDataSnapshot alloc]
-                       initWithRef:query.ref
-                       indexedNode:[FIndexedNode
-                                       indexedNodeWithNode:node
-                                                     index:querySpec.index]]);
+        dispatch_async(self.config.callbackQueue, ^{
```

Comment:
Can you use `eventRaiser`? https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseDatabase/Sources/Core/FRepo.m#L184